### PR TITLE
Unifies input as JSON and Validates it

### LINKED ISSUE #6: Unify input as JSON
Create a module that reads a YAML or JSON string, and converts the input to JSON. If the input is already JSON, it should perform no conversion.

- [ ] :bug: Unit tests
- [ ] :books: Documentation

### LINKED ISSUE #9: Validate JSON input
Create a module that takes a JSON string and validates it using a JSON Schema Draft 07 validator. It should return 2 things: `validatedDocument, err`.

- [ ] :bug: Unit tests
- [ ] :books: Documentation

### PR DESCRIPTION
**Pull request by fmvilas**
### What?

Fixes #6
Fixes #9

### CODE REVIEW

File: hlsp/hlsp.go [hunk 1]
```diff
@@ -1,46 +1,45 @@
...
+	"encoding/json"
 
+	"github.com/ghodss/yaml"
 	"github.com/xeipuuv/gojsonschema"
 )
 
-// Parse receives either a YAML or JSON AsyncAPI document.
+// Parse receives either a YAML or JSON AsyncAPI document, and tries to parse it.
+func Parse(yamlOrJSONDocument []byte) (json.RawMessage, *ParserError) {
+	jsonDocument, err := yaml.YAMLToJSON(yamlOrJSONDocument)
```
Comment (rmelian) [inline hlsp/hlsp.go]: the call to YAMLToJSON looks like its always doing a conversion to JSON from yaml but it is not. it only does it if the content is json.

Comment (fmvilas) [reply hlsp/hlsp.go]: What do you mean? It's being tested here and it works: https://github.com/asyncapi/parser/pull/18/files#diff-fee0983ac72dab9799141e78d9c594f9R47

Comment (rmelian) [reply hlsp/hlsp.go]: sorry, I meant the name, but I guess I'm been too picky  :)

Comment (fmvilas) [reply hlsp/hlsp.go]: I'm not understanding 😅

Comment (rmelian) [reply hlsp/hlsp.go]: haha, no worries its fine. this looks good to me

**Pull request merged**